app/src/main/java/com/island/island/Activities/EditProfileActivity.java

```diff
@@ -10,72 +11,115 @@
-    {
-        if (resultCode == RESULT_OK)
-        {
-            if (requestCode == SELECT_PROFILE_IMAGE)
-            {
-                // TODO: set profile image and save uri to database
-                Uri selectedImageUri = data.getData();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode == RESULT_OK) {
+            if (requestCode == SELECT_PROFILE_IMAGE) {
+                profileImageUri = data.getData();
+                ImageUtils.setProfileImageSampled(getApplicationContext(),
+                        profileImage, profileImageUri);
+
             }
-            else if(requestCode == SELECT_HEADER_IMAGE)
-            {
-                // TODO: set header image
-                Uri selectedImageUri = data.getData();
+            else if(requestCode == SELECT_HEADER_IMAGE) {
+                headerImageUri = data.getData();
+                ImageUtils.setHeaderImageSampled(getApplicationContext(),
+                        headerImage, headerImageUri);
             }
         }
     }
 
-    public void saveProfile(View view)
-    {
-        // TODO: handle image stuff for saving
+    public void saveProfile(View view) {
         String newAboutMeText = aboutMe.getText().toString();
         String myUsername = PreferenceManager.getDefaultSharedPreferences(getApplicationContext())
                 .getString(getApplicationContext().getString(R.string.user_name), "");
-        Profile newProfile = VersionedContentBuilder.buildProfile(getApplicationContext(), myUsername, newAboutMeText);
+
+        // TODO: Implement separate REST calls for profile/header images
+        Uri newProfileUri = profileImageUri == null ? prevProfileImageUri : profileImageUri;
+        Uri newHeaderUri = headerImageUri == null ? prevHeaderImageUri : headerImageUri;
+
+        ProfileWithImageData newProfileWithImageData = VersionedContentBuilder.buildProfile(
+                getApplicationContext(),
+                myUsername,
+                newAboutMeText,
+                ImageUtils.getByteArrayFromUri(getApplicationContext(), newProfileUri),
+                ImageUtils.getByteArrayFromUri(getApplicationContext(), newHeaderUri)
+        );
+
+        // TODO: This saves a new image every time. Will change with new REST calls.
+        Uri savedProfileImageUri = ImageUtils.saveBitmapToInternalFromUri(getApplicationContext(),
+                newProfileUri);
+        Uri savedHeaderImageUri = ImageUtils.saveBitmapToInternalFromUri(getApplicationContext(),
+                newHeaderUri);
+
+        Profile newProfile = new Profile(
+                myUsername,
+                newAboutMeText,
+                savedProfileImageUri,
+                savedHeaderImageUri,
+                newProfileWithImageData.getVersion()
+        );
+
         ProfileDatabase.getInstance(getApplicationContext()).update(newProfile);
```

Comment:
I agree. This can probably wait until we get the new Image REST calls implemented.